rfcs/text/0430-hook-service.md

```diff
@@ -0,0 +1,148 @@
+
+An embedding application receiving a `hook` message should use this to directly manipulate `node` in order to render the `type` view for the specified `path`.
+
+Where a hook makes use of data, the embedding application should normally read this from the `path` specified in the hook, and ensure the data available at this `path` is updated as changes are made. It may be that the hook stores data in a different format that the block's schema specifies. In this case, the embedding application should still, where possible, ensure a valid representation (i.e, fits the type specified in the block's schema) of this data is available at the relevant `path`.
+
+When receiving `null` for the `node`, the embedding application should tear down any and all subscriptions made in association with this hook.
+
+As this service relies on the embedding application directly manipulating DOM nodes, it will not be possible to implement this where block's 'proxy' the Block Protocol using `postMessage` (i.e, where a block includes an `iframe`). This service requires that the embedding application implementation of the service share the global environment (commonly referred to as a 'scope' or 'realm') with the DOM node passed by the block in the `hook` message.
+
+### Errors
+
+Where an embedding application has not implemented a hook, it must respond with a `NOT_IMPLEMENTED` error, which will allow blocks to implement a fallback. This error must include the `node`, `type`, `path` and `hookId` properties passed in the original `hook` message.
+
+### Notes
+
+- This RFC makes no changes to the Graph Service's requirements that an embedding application implementing the Graph Service must provide data in the format specified by a block's schema.
+- The embedding application remains in control of data, as specified in the Graph Service, and should continue to indicate whether the block is in a readonly state as specified in that service. The implementation of the Hook Service should mirror this state.
+
+## Core Specification changes
+
+When an embedding application receives a message from a block where it does the implement the `service` specified in the message, it must respond with a `NOT_IMPLEMENTED` error, so that blocks can respond appropriately.
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+TODO:
+
+This is the technical portion of the RFC. Explain the design in sufficient detail that:
+
+- Its interaction with other features is clear.
+- It is reasonably clear how the feature would be implemented.
+- Corner cases are dissected by example.
+
+The section should return to the examples given in the previous section, and explain more fully how the detailed proposal makes those examples work.
+
```

Comment:
Suggest using the content above as the reference-level, and therefore moving this up / deleting the template content
```suggestion
```